### PR TITLE
Rename components

### LINKED ISSUE #1871: Move components names to `Ruler`, `Compactor`, `Querier` (`-er`)
It's maybe a minor but we should be consistent on naming. In the documentation, code and `thanos` command we have `rule, compact, downample, query` vs in some docs, deployments and most of the presentations we are leaking `Querier`, `Ruler`, `Compactor` 

This conflicts a bit with `Store Gateway` but we already have some discussion about renaming it: https://github.com/thanos-io/thanos/issues/1646

I would vote `-er` as it's more correct (aka `go` way of naming objects as well) and it avoids name collisions like `rule` (recording/alerting rule) vs `rule` (the component responsible for evaluating rules). 

**AC**: 
 < to agree>

Any thoughts?

### PR DESCRIPTION
**Pull request by yeya24**
* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

Fixes #1871

### CODE REVIEW

**Comment by yeya24** [comment]
Forgot this. I will rebase it tomorrow!

**Comment by bwplotka** [comment]
@yeya24, before you rebase have you seen our final decision about naming? https://thanos.io/contributing/#components-naming

**Comment by kakkoyun** [comment]
@yeya24 Correct me if I'm wrong, but the decision was actually to keep command names as is in the code, no?

**Comment by yeya24** [comment]
@bwplotka  @kakkoyun 
Thanks! Seems I need to update my PR first!
Just to confirm what place should I change:
- Docs
- Code comments

And a small question: What about `receive`? I don' t see it in https://thanos.io/contributing/#components-naming. Do I need to rename it to `receiver`?

**Comment by kakkoyun** [comment]
@yeya24 It should be `receive` as well, as in command. `thanos receive`. We can also update the that in the document.

**Comment by yeya24** [comment]
> @yeya24 It should be `receive` as well, as in command. `thanos receive`. We can also update the that in the document.

Thanks. I will pin you after I finish.

File: docs/design.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ The meta.json is updated during upload time on sidecars.
 
 ```
 ┌────────────┬─────────┐         ┌────────────┬─────────┐     ┌─────────┐
-│ Prometheus │ Sidecar │   ...   │ Prometheus │ Sidecar │     │   Rule  │
+│ Prometheus │ Sidecar │   ...   │ Prometheus │ Sidecar │     │  Ruler  │
```
Comment (yeya24) [inline docs/design.md]: Maybe we need to update the graph to add `Receiver`

**Comment by yeya24** [comment]
I think it is ready. Based on https://thanos.io/contributing/#components-naming, most changes are just renaming in docs.

**Pull request closed without merging**